AirCasting/Dashboard/DashboardView.swift

```diff
@@ -37,58 +48,97 @@ struct DashboardView: View {
         .onChange(of: selectedSection.selectedSection) { selectedSection in
             self.selectedSection.selectedSection = selectedSection
             try! coreDataHook.setup(selectedSection: self.selectedSection.selectedSection)
         }
+        .onChange(of: isRefreshing, perform: { newValue in
+            guard newValue == true else { return }
+            guard !sessionSynchronizer.syncInProgress.value else {
+                onCurrentSyncEnd { isRefreshing = false }
+                return
+            }
+            sessionSynchronizer.triggerSynchronization() { isRefreshing = false }
+        })
         .onAppear() {
             try! coreDataHook.setup(selectedSection: self.selectedSection.selectedSection)
         }
     }
     
+    private var sessionTypePicker: some View {
+        AirSectionPickerView(selection: self.$selectedSection.selectedSection)
+            .padding(.leading)
+            .background(
+                ZStack(alignment: .bottom) {
+                    Color.green
+                        .frame(height: 3)
+                        .shadow(color: Color.aircastingDarkGray.opacity(0.4),
+                                radius: 6)
+                        .padding(.horizontal, -30)
+                    Color.white
+                }
+            )
+            .zIndex(2)
+    }
+    
+    private var emptySessionsView: some View {
+        GeometryReader { geometry in
+            ScrollView(.vertical, showsIndicators: false) {
+                VStack {
+                    RefreshControl(coordinateSpace: .named(dashboardCoordinateSpaceName), isRefreshing: $isRefreshing)
+                    if selectedSection.selectedSection == .mobileActive || selectedSection.selectedSection == .mobileDormant {
+                        EmptyMobileDashboardViewMobile()
+                            .frame(height: geometry.size.height)
+                    } else {
+                        EmptyFixedDashboardView()
+                            .frame(height: geometry.size.height)
+                    }
+                }
+            }
+            .coordinateSpace(name: dashboardCoordinateSpaceName)
+            .background(Color.aliceBlue)
+        }
+    }
+    
+    private var sessionListView: some View {
+        ZStack(alignment: .bottomTrailing) {
+            Image("dashboard-background-thing")
+            let thresholds = Array(self.thresholds)
+            ScrollView {
+                RefreshControl(coordinateSpace: .named(dashboardCoordinateSpaceName), isRefreshing: $isRefreshing)
+                LazyVStack(spacing: 8) {
+                    ForEach(sessions.filter { $0.uuid != "" && !$0.gotDeleted }, id: \.uuid) { session in
+                        let followingSetter = MeasurementStreamStorageFollowingSettable(session: session, measurementStreamStorage: measurementStreamStorage)
+                        let viewModel = SessionCartViewModel(followingSetter: followingSetter)
```

Comment:
should be SessionCardViewModel, right? I'm always confused by these one-letter differences so ignore me if it's correct :D

Reply:
You are totally right. I am not great at naming well 😅